regulations/tasks.py

```diff
@@ -152,3 +164,18 @@ def make_s3_client():
         aws_secret_access_key=settings.ATTACHMENT_SECRET_ACCESS_KEY,
     )
     return session.client('s3', config=Config(signature_version='s3v4'))
+
+
+def extract_files(body):
+    '''
+    Extracts the files that are to be attached to the comment.
+    Returns a collection of dicts where for each dict:
+        - dict['key'] specifies the file to be attached from S3
+        - dict['name'] specifies the name under which the file is to be
+          attached.
+    '''
+    return [
+        file
+        for section in body.get('sections', [])
```

Comment:
List is right. A typical submission is:
```
{"assembled_comment": {
    "sections" : [
    {"id": "A1", "comment": "A simple comment", "files": [{"name": "doc0.pdf", "key": "doc0.pdf"}]}
    {"id": "A5", "comment": "Another comment", "files": [{"name": "doc1.pdf", "key": "doc2.pdf"}, {"name": "doc1.pdf", "key": "doc2.pdf"}]}
]},
"submitter_rep": "The Advocate"}
```

I would be OK with removing the intermediary `sections` as it adds no value.